NExtends/Expressions/ExpressionHelper.cs

```diff
@@ -0,0 +1,236 @@
+        {
+            return BuildEnumerableMethod(tSource, accessor, predicate, "Where");
+        }
+
+        private static Expression BuildEnumerableMethod(Type tSource, Expression accessor, Expression predicate, string method)
+        {
+            var overload = typeof(Enumerable).GetMethods()
+                                      .Single(mi => mi.Name == method && mi.GetParameters().Count() == 2).MakeGenericMethod(tSource);
+
+            var call = Expression.Call(
+                overload,
+                accessor,
+                predicate);
+
+            return call;
+        }
+
+        public static bool HasSubexpressionsOfType<TTest>(Expression original)
+        {
+            var visitor = new HasTypeVisitor<TTest>();
+            return visitor.VisitRoot(original);
+        }
+
+        private class HasTypeVisitor<TExpression> : ExpressionVisitor
+        {
+            private readonly Type _searchedType;
+            private bool _HasType = false;
+
+            public HasTypeVisitor()
+            {
+                _searchedType = typeof(TExpression);
+            }
+
+            public bool VisitRoot(Expression node)
+            {
+                Visit(node);
+                return _HasType;
+            }
+
+            public override Expression Visit(Expression node)
+            {
+                if (_HasType) { return node; }
+                if (node != null && node.Type == _searchedType)
+                {
+                    _HasType = true;
+                    return node;
+                }
+                else
+                {
+                    return base.Visit(node);
+                }
+            }
+        }
+
+        public ICollection<Expression<Func<TEntity, object>>> Remove<TEntity>(IEnumerable<Expression<Func<TEntity, object>>> source, Expression<Func<TEntity, object>> candidate)
+        {
+            var result = new HashSet<Expression<Func<TEntity, object>>>();
+
+            foreach (var exp in source)
+            {
+                if (!Equals(exp, candidate))
+                {
+                    result.Add(exp);
+                }
+            }
+
+            return result;
+        }
+
+        public bool Equals<TEntity>(Expression<Func<TEntity, object>> exp1, Expression<Func<TEntity, object>> exp2)
+        {
+            return GetPropertyFromPropertySelector<TEntity>(exp1) == GetPropertyFromPropertySelector<TEntity>(exp2);
+        }
+
+        /// <summary>
+        /// Récupère la propriété ciblée par une expression, notamment pour voir si 2 expressions sont égales = récupèrent la même propriété
+        /// </summary>
+        /// <typeparam name="TEntity"></typeparam>
+        /// <param name="propertySelector"></param>
+        /// <returns></returns>
+        private PropertyInfo GetPropertyFromPropertySelector<TEntity>(Expression<Func<TEntity, object>> propertySelector)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'System.Linq.Expressions.LambdaExpression' instead of 'System.Linq.Expressions.Expression<System.Func<TEntity, object>>'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3242)